src/lib.rs

```diff
@@ -28,8 +28,12 @@ mod testing;
 mod units;
 
 pub trait DataIO<Data> {
-    type Error: Debug;
+    type Error: Debug + 'static;
     fn get_data(&self) -> Data;
+    fn check_availability(
```

Comment:
Please add a comment explaining the semantics of `Option` here – i.e. (if I understood correctly) that returning `None` means "don't worry about availability, everything is always fine".

Reply:
done